tests/Integration/Behaviour/Features/Context/Domain/AttributeGroupFeatureContext.php

```diff
@@ -30,15 +30,116 @@
 
 use Behat\Gherkin\Node\TableNode;
 use PHPUnit\Framework\Assert;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Command\AddAttributeGroupCommand;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Command\EditAttributeGroupCommand;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Exception\AttributeGroupConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Query\GetAttributeGroupForEditing;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\ValueObject\AttributeGroupId;
+use PrestaShop\PrestaShop\Core\Domain\Product\AttributeGroup\Command\DeleteAttributeGroupCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\AttributeGroup\Exception\AttributeGroupNotFoundException;
 use PrestaShop\PrestaShop\Core\Domain\Product\AttributeGroup\Query\GetAttributeGroupList;
 use PrestaShop\PrestaShop\Core\Domain\Product\AttributeGroup\Query\GetProductAttributeGroups;
 use PrestaShop\PrestaShop\Core\Domain\Product\AttributeGroup\QueryResult\AttributeGroup;
 use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
 use RuntimeException;
+use Tests\Integration\Behaviour\Features\Context\CommonFeatureContext;
+use Tests\Integration\Behaviour\Features\Context\SharedStorage;
+use Tests\Integration\Behaviour\Features\Context\Util\NoExceptionAlthoughExpectedException;
 use Tests\Integration\Behaviour\Features\Context\Util\PrimitiveUtils;
 
 class AttributeGroupFeatureContext extends AbstractDomainFeatureContext
 {
+    /**
+     * @var int default language id from configs
+     */
+    private $defaultLangId;
+
+    /**
+     * @var int default shop id from configs
+     */
+    private $defaultShopId;
+
+    public function __construct()
+    {
+        $configuration = CommonFeatureContext::getContainer()->get('prestashop.adapter.legacy.configuration');
+        $this->defaultLangId = $configuration->get('PS_LANG_DEFAULT');
+        $this->defaultShopId = $configuration->get('PS_SHOP_DEFAULT');
+    }
+
+    /**
+     * @When I create attribute group :reference with specified properties:
+     */
+    public function createAttributeGroup(string $reference, TableNode $node): void
+    {
+        $properties = $node->getRowsHash();
+        $attributeGroupId = $this->createAttributeGroupUsingCommand($properties['name'], $properties['public_name'], $properties['type']);
+
+        $this->getSharedStorage()->set($reference, $attributeGroupId->getValue());
+    }
+
+    /**
+     * @When I edit attribute group :reference with specified properties:
+     */
+    public function editAttributeGroup(string $reference, TableNode $node): void
+    {
+        $attributeGroupId = SharedStorage::getStorage()->get($reference);
```

Comment:
You shouldn't use the static calls any more as we have a better helper method for this
```suggestion
        $attributeGroupId = $this->getStorage()->get($reference);
```
SAme thing when you set the reference after creation, you should replace all the call to `SharedStorage::getStorage()` with `$this->getSharedStorage()`

Reply:
But for references, we even have an even more adapted helper:
```suggestion
        $attributeGroupId = $this->referenceToId($reference);
```